spec/models/alaveteli_pro/plan_with_discount_spec.rb

```diff
@@ -0,0 +1,43 @@
+# -*- encoding : utf-8 -*-
+require 'spec_helper'
+
+describe AlaveteliPro::PlanWithDiscount do
+  let(:plan) { OpenStruct.new(amount: 833) }
+  let(:subscription) { OpenStruct.new(plan: plan) }
+  subject { described_class.new(subscription) }
+
+  describe '#amount' do
+
+    context 'no discount is set' do
+      it 'returns the original stripe plan amount' do
+        expect(subject.amount).to eq(833)
+      end
+    end
+
+    it 'applies a percentage discount correctly' do
+      coupon = OpenStruct.new(id: "50_off", object: "coupon", amount_off: nil,
+                              duration: "forever", max_redemptions: nil,
+                              percent_off: 50, redeem_by: nil,
+                              times_redeemed: 1, valid: true)
+      discount = OpenStruct.new(coupon: coupon)
+      subscription = OpenStruct.new(plan: plan, discount: discount)
+      subject = described_class.new(subscription)
+
+      expect(subject.amount).to eq(416.5)
```

Comment:
grudgingly fixing this to match Stripe as this was supposed to be a calculation value that would always have a tax calculation applied rather than a cash amount to go to Stripe (or anywhere else) - too used to seeing my utility bills use fractions of pence to make sure I don't underpay them

Reply:
> this was supposed to be a calculation value that would always have a tax calculation applied 

That's wrong for dealing with invoices:

> **Don't show VAT amounts for individual items**
> 
> Fiscal authorities do not require you to show the amount of VAT for each individual invoice item. All you need is a proper VAT total for the complete invoice.
> 
> Because of the rounding rules above, it is also extremely difficult to show VAT amounts for individual amounts while still making these numbers add up to the total VAT. Don't go there.
>
> – _https://makandracards.com/makandra/1505-invoices-how-to-properly-round-and-calculate-totals#dont-show-vat-amounts-for-individual-items_

A subscription is a line-item in an order, so yeah, should be without VAT.